lib/cldr.js

```diff
@@ -490,6 +490,34 @@ Cldr.prototype = {
     return territoryDisplayNames;
   },
 
+  /**
+   * Subdivisions are usually subnational administrative entities.
+   * Codes follow the BCP47 standard, e.g. `usca` for California, USA.
+   * Note that these codes are similar but not identical to ISO 3166-2 codes.
+   * Unlike ISO 3166-2, CLDR never reuses a code.
+   *
+   * @param {string} localeId Locale or subdivision names
+   * @returns {Object} A dictionary of subdivision codes and their names.
+   *
+   * @example
+   * // returns 'Zealand'
+   * cldr.extractSubdivisionDisplayNames('en').dk85
+   */
+  extractSubdivisionDisplayNames(localeId) {
+    this.checkValidLocaleId(localeId);
+    const finder = this.createFinder(
+      this.getPrioritizedDocumentsForLocale(localeId, 'subdivisions')
+    );
+    const subdivisionDisplayNames = {};
+    finder('/ldml/localeDisplayNames/subdivisions/subdivision').forEach(node => {
+      if (node.getAttribute('alt') !== 'variant') {
```

Comment:
Other extractors rely on the fact that the “most correct” one always comes first. I think `alt` has other possible values that also indicates less certainty?